### PR TITLE
fix: RabbitMQ 3.13 and ActiveMQ 5.18 Version Mismatch

### LINKED ISSUE #38708: [Bug]: modifying the configuration of an ActiveMQ broker running 5.18 fails with BadRequestException: Broker engine version [5.18.4] is invalid for broker engine type [ACTIVEMQ]
### Terraform Core Version

1.4.2, OpenTofu 1.6.2

### AWS Provider Version

5.61.0

### Affected Resource(s)

* aws_mq_broker
* aws_mq_configuration

### Expected Behavior

The plan from Terraform does not modify the `engine_version`, only the configuration.

### Actual Behavior

The engine version is configured in Terraform as "5.18", but reported by AWS as "5.18.4". My assumption is that the provider tries to re-send the reported version back to AWS, which then fails.

### Relevant Error/Panic Output Snippet

```shell
Error: updating MQ Broker (XXX) configuration: BadRequestException: Broker engine version [5.18.4] is invalid for broker engine type [ACTIVEMQ]. Valid values: [5.18, 5.17.6, 5.16.7, 5.15.16].│ {│   RespMetadata: {│     StatusCode: 400,│     RequestID: "cffc2472-2dcf-4062-a2e9-2ded71335611"│   },│   ErrorAttribute: "engineVersion",│   Message_: "Broker engine version [5.18.4] is invalid for broker engine type [ACTIVEMQ]. Valid values: [5.18, 5.17.6, 5.16.7, 5.15.16]."│ }│ │   with module.mq_broker.aws_mq_broker.default,│   on ../modules/ctd-activemq/mq.tf line 11, in resource "aws_mq_broker" "default":│   11: resource "aws_mq_broker" "default" {│ ╵
```

### Terraform Configuration Files
```terraform
resource "aws_security_group" "example" {
  name = "example"

  ingress {
    description      = "MQTT over TLS"
    from_port        = 8883
    to_port          = 8883
    protocol         = "tcp"
    cidr_blocks      = ["0.0.0.0/0"]
    ipv6_cidr_blocks = ["::/0"]
  }

  egress {
    from_port        = 0
    to_port          = 0
    protocol         = "-1"
    cidr_blocks      = ["0.0.0.0/0"]
    ipv6_cidr_blocks = ["::/0"]
  }
}

# resource "aws_mq_configuration" "example" {
#   description    = "Example Configuration"
#   name           = "example"
#   engine_type    = "ActiveMQ"
#   engine_version = "5.18"

#   data = <<DATA
# <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
# <broker xmlns="http://activemq.apache.org/schema/core">
# </broker>
# DATA
# }

resource "aws_mq_broker" "example" {
  broker_name = "example"

  engine_type                = "ActiveMQ"
  engine_version             = "5.18"
  host_instance_type         = "mq.t3.micro"
  apply_immediately          = true
  auto_minor_version_upgrade = true
  security_groups            = [aws_security_group.example.id]

  # configuration {
  #   id       = aws_mq_configuration.example.id
  #   revision = aws_mq_configuration.example.latest_revision
  # }

  lifecycle {
    ignore_changes = [engine_version]
  }

  user {
    username = "ExampleUser"
    password = "LongPasswordString"
  }
}
```

### Steps to Reproduce

1. Run an initial apply
2. Un-comment the `aws_mq_configuration` block and the configuration in `aws_mq_broker`
3. Run another apply

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

### LINKED ISSUE #38930: [Bug]: aws_mq_broker only allows me to set engine version to 3.13 instead of 3.13.6
### Terraform Core Version

1.8.2-1

### AWS Provider Version

5.63.0

### Affected Resource(s)

aws_mq_broker

### Expected Behavior

Allowed me to set engine version using the minor version too.

### Actual Behavior

Keeps trying to set it to 3.13

```
# module.aws.module.rabbitmq_v3_13.aws_mq_broker.broker will be updated in-place
~ resource "aws_mq_broker" "broker" {
    ~ engine_version                = "3.13.6" -> "3.13"
      id                            = "b-e380aa62-9e30-4f2e-bac5-bffb1415307c"
      tags                          = {
          "App" = "Shared"
      }
      # (16 unchanged attributes hidden)

    ~ configuration {
        ~ id       = "c-5902d33e-0ef9-45d1-b5c2-abb62fa27812" -> "c-917a03fc-aaed-4f9c-9f5c-2e92e669376e"
          # (1 unchanged attribute hidden)
      }

      # (3 unchanged blocks hidden)
  }
```

But later, on the next apply, it notices AWS set it to 3.13.6

```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply" which may have affected this plan:

  # module.aws.module.rabbitmq_v3_13.aws_mq_broker.broker has changed
  ~ resource "aws_mq_broker" "broker" {
      ~ engine_version                = "3.13" -> "3.13.6"
        id                            = "b-e380aa62-9e30-4f2e-bac5-bffb1415307c"
        tags                          = {
            "App" = "Shared"
        }
        # (16 unchanged attributes hidden)

      ~ configuration {
          ~ id       = "c-917a03fc-aaed-4f9c-9f5c-2e92e669376e" -> "c-5902d33e-0ef9-45d1-b5c2-abb62fa27812"
            # (1 unchanged attribute hidden)
        }

        # (3 unchanged blocks hidden)
    }
```

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

If I try to set to 3.13.6, it gives an error.

```
│ Error: creating MQ Configuration (staging-main-v3_13-configuration): operation error mq: CreateConfiguration, https response error StatusCode: 400, RequestID: 14095f86-a3c7-4185-9413-7bc6f95c873e, BadRequestException: Broker engine version [3.13.6] is invalid for broker engine type [RABBITMQ]. Valid values: [3.13, 3.12.13, 3.11.28, 3.11.20, 3.11.16, 3.10.25, 3.10.20, 3.10.10, 3.9.27].
```

### Steps to Reproduce

Try to create any rabbitmq instance with the version 3.13.6

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by architec**
### Description

Terraform encounters version mismatch with Amazon MQ RabbitMQ 3.13 and ActiveMQ 5.18, causing unnecessary reboots due to patch-level updates. 

### Relations

Closes #38930
Closes #38708

### References

> Starting from RabbitMQ 3.13, Amazon MQ will manage patch version upgrades for your brokers. All brokers on version 3.13 will be automatically upgraded to the latest compatible and secure patch version in your scheduled maintenance window.

https://aws.amazon.com/about-aws/whats-new/2024/07/amazon-mq-rabbitmq-version-3-13/

### Output from Acceptance Testing

```console
% PKG=mq make testacc TESTS=TestAccMQBroker_basic

% PKG=mq make testacc TESTS=TestAccMQBroker_RabbitMQ_basic

% PKG=mq make test
...
```

### CODE REVIEW

File: internal/service/mq/broker.go
Comment (architec) [reply internal/service/mq/broker.go:471]: Indeed, this PR will necessitate code updates when Amazon MQ releases new versions. However, I can submit a subsequent pull request to modify the change, making it applicable to all versions >= 3.13 for RabbitMQ and versions >= 5.18 for ActiveMQ.

With the subsequent PR, adding an attribute such as `actual_engine_version` is unnecessary.

**Comment by jar-b** [comment]
Thanks for your submission @architec! 👍 

I'm working through some failures on the RabbitMQ acceptance tests which appear unrelated to this change (tags drift). Once those are resolved I'll proceed with merging this in.

**Comment by jar-b** [comment]
I've isolated the acceptance testing failures to a change in Terraform core `1.8.0`. Running with `1.7.5` yields passing tests, so I'm going to run the full suite broker suite with that version now and file a separate issue to continue this research. 

Will merge this once the full test suite completes.

```console
% TF_ACC_TERRAFORM_VERSION=1.7.5 make testacc PKG=mq TESTS="TestAccMQBroker_normalizedEngineVersion|TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade|TestAccMQBroker_RabbitMQ_normalizedEngineVersion"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/mq/... -v -count 1 -parallel 20 -run='TestAccMQBroker_normalizedEngineVersion|TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade|TestAccMQBroker_RabbitMQ_normalizedEngineVersion'  -timeout 360m
=== RUN   TestAccMQBroker_normalizedEngineVersion
=== PAUSE TestAccMQBroker_normalizedEngineVersion
=== RUN   TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade
=== PAUSE TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade
=== RUN   TestAccMQBroker_RabbitMQ_normalizedEngineVersion
=== PAUSE TestAccMQBroker_RabbitMQ_normalizedEngineVersion
=== CONT  TestAccMQBroker_normalizedEngineVersion
=== CONT  TestAccMQBroker_RabbitMQ_normalizedEngineVersion
=== CONT  TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/index.json
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/index.json
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/index.json
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS.72D7468F.sig
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS.72D7468F.sig
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS.72D7468F.sig
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_SHA256SUMS
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_darwin_arm64.zip
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_darwin_arm64.zip
2024/09/12 11:43:45 [DEBUG] GET https://releases.hashicorp.com/terraform/1.7.5/terraform_1.7.5_darwin_arm64.zip
--- PASS: TestAccMQBroker_RabbitMQ_normalizedEngineVersion (681.96s)
--- PASS: TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade (686.64s)
--- PASS: TestAccMQBroker_normalizedEngineVersion (1182.01s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/mq 1188.204s
```

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% TF_ACC_TERRAFORM_VERSION=1.7.5 make testacc PKG=mq TESTS="TestAccMQBroker_"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/mq/... -v -count 1 -parallel 20 -run='TestAccMQBroker_'  -timeout 360m

--- PASS: TestAccMQBroker_RabbitMQ_validationAuditLog (631.07s)
=== CONT  TestAccMQBroker_normalizedEngineVersion
--- PASS: TestAccMQBroker_RabbitMQ_normalizedEngineVersion (647.54s)
=== CONT  TestAccMQBroker_RabbitMQ_basic
--- PASS: TestAccMQBroker_RabbitMQ_autoMinorVersionUpgrade (659.43s)
=== CONT  TestAccMQBroker_Update_hostInstanceType
--- PASS: TestAccMQBroker_RabbitMQ_config (662.56s)
--- PASS: TestAccMQBroker_RabbitMQ_cluster (669.10s)
--- PASS: TestAccMQBroker_RabbitMQ_logs (672.09s)
--- PASS: TestAccMQBroker_throughputOptimized (939.31s)
--- PASS: TestAccMQBroker_ldap (953.94s)
--- PASS: TestAccMQBroker_EncryptionOptions_kmsKeyID (989.45s)
--- PASS: TestAccMQBroker_EncryptionOptions_managedKeyEnabled (1038.29s)
--- PASS: TestAccMQBroker_EncryptionOptions_managedKeyDisabled (1043.15s)
--- PASS: TestAccMQBroker_disappears (1060.67s)
--- PASS: TestAccMQBroker_basic (1064.66s)
--- PASS: TestAccMQBroker_tags (1078.43s)
--- PASS: TestAccMQBroker_Update_engineVersion (1240.34s)
--- PASS: TestAccMQBroker_Update_securityGroup (1262.55s)
--- PASS: TestAccMQBroker_RabbitMQ_basic (660.77s)
--- PASS: TestAccMQBroker_Update_users (1429.44s)
--- PASS: TestAccMQBroker_normalizedEngineVersion (843.55s)
--- PASS: TestAccMQBroker_AllFields_defaultVPC (1593.85s)
--- PASS: TestAccMQBroker_AllFields_customVPC (1740.74s)
--- PASS: TestAccMQBroker_Update_hostInstanceType (1353.62s)
--- PASS: TestAccMQBroker_dataReplicationMode (2095.33s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/mq 2101.590s
```

**Comment by jar-b** [comment]
Thanks for your contribution, @architec! 👍 

Apologies for the long running review.

**Pull request merged**
